drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -284,16 +284,27 @@ def self.generate_enrollment_id
   def save_new_enrollment!
     raise 'Unexpected: save_new_enrollment called on a persisted enrollment' if persisted?
 
-    if Hmis::ProjectAutoEnterConfig.detect_best_config_for_project(project)
-      # If auto-enter is configured for this project, save as non-WIP and generate an empty intake.
-      save_not_in_progress!
-      build_synthetic_intake_assessment.save!
+    if should_auto_enter?
+      save_and_auto_enter!
     else
-      # Otherwise, save as WIP
       save_in_progress!
     end
   end
 
+  def should_auto_enter?
+    Hmis::ProjectAutoEnterConfig.detect_best_config_for_project(project)
```

Comment:
Methods with `?` by convention should return a boolean (can tack on `.present?`)